presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeSearcher.java

```diff
@@ -31,12 +31,15 @@
 
 public class PlanNodeSearcher
 {
-    @Deprecated
     public static PlanNodeSearcher searchFrom(PlanNode node)
     {
         return searchFrom(node, noLookup());
     }
 
+    /**
+     * Use it in optimizer {@link com.facebook.presto.sql.planner.iterative.Rule} if you truely do not have a better option
+     */
+    @Deprecated
```

Comment:
Then TODO would be more proper than deprecation. And make it deprecation once we have alternative.